data/provider/src/test/java/tech/pegasys/teku/api/ConfigProviderTest.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.api.response.v1.config.GetSpecResponse;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecFactory;
+import tech.pegasys.teku.spec.config.SpecConfig;
+import tech.pegasys.teku.spec.config.SpecConfigLoader;
+
+class ConfigProviderTest {
```

Comment:
Should we add also a negative test when the config provider fail to load?

Reply:
I'm not exactly sure what you mean. You can't start without having a `Spec` object so there shouldn't ever be a case where we can't generate the config response and there shouldn't ever be a case where we can't parse that output again.

We do have a bunch of tests in `SpecConfigReaderTest` for how it handles invalid configs - the aim here is just to check that the config we get from our REST API is considered valid.

Reply:
All good, it was a good question to ask.